engine/src/main/java/org/terasology/rendering/nui/layers/ingame/metrics/DebugMetricsSystem.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2016 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.terasology.rendering.nui.layers.ingame.metrics;
+
+
+import com.google.common.collect.Iterators;
+import com.google.common.collect.Sets;
+import org.terasology.entitySystem.systems.BaseComponentSystem;
+import org.terasology.entitySystem.systems.RegisterSystem;
+import org.terasology.registry.Share;
+
+import java.util.Iterator;
+import java.util.LinkedHashSet;
+
+/**
+ * Provides MetricsMode objects to DebugOverlay. Default metrics nodes can be modified via register
+ * unregister and unregisterAll functions.
```

Comment:
Good start. I'd write:

/**
Manages an ordered set of MetricsMode instances. 

A number of default metrics modes is instantiated in the initialize() method and becomes immediately available by iterating through the set repeatedly invoking the toggle() method. The getCurrentMode() method can also be used to obtain the mode currently pointed at. Further modes can be added to the set via the register method, while registered modes can be removed via the unregister method. A convenience method unregisterAll() removes all registered modes at once.

MetricsModes are currently used by the DebugOverlay instance, displaying runtime statistics on screen.
*/

Notice that I'm being extra-verbose because eventually these methods will become part of the API, available to other parts of the engine and third-party modules. In fact, we might want to do it right away. How does this work @Cervator? How do we register these methods with the API? Any quirk we should know?